homeassistant/components/amcrest/__init__.py

```diff
@@ -298,13 +302,13 @@ def setup(hass: HomeAssistant, config: ConfigType) -> bool:
             control_light,
         )
 
-        discovery.load_platform(
+        await discovery.async_load_platform(
```

Comment:
When loading the platforms, we should create a task to avoid a deadlock. Please use `hass.async_create_task`.